attestation-agent/kbs_protocol/src/client/passport.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{bail, Result};
+use async_trait::async_trait;
+use kbs_types::{ErrorInformation, Response};
+use log::{debug, warn};
+use resource_uri::ResourceUri;
+
+use crate::{
+    api::KbsClientCapabilities,
+    client::{KbsClient, KBS_GET_RESOURCE_MAX_ATTEMPT, KBS_PREFIX},
+    token_provider::TokenProvider,
+};
+
+impl KbsClient<Box<dyn TokenProvider>> {
+    async fn update_token(&mut self) -> Result<()> {
+        let (token, teekey) = self.provider.get_token().await?;
+        self.token = Some(token);
+        self.tee_key = teekey;
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl KbsClientCapabilities for KbsClient<Box<dyn TokenProvider>> {
```

Comment:
I thought the point of the passport model was to return a Token, but we aren't implementing `PassportClientCapabilities` here at all. Am I missing something?

Reply:
see https://github.com/confidential-containers/guest-components/pull/304#discussion_r1281589440